src/cluster.c

```diff
@@ -1488,6 +1497,7 @@ void clusterProcessGossipSection(clusterMsg *hdr, clusterLink *link) {
                 if (node->link) freeClusterLink(node->link);
                 memcpy(node->ip,g->ip,NET_IP_STR_LEN);
                 node->port = ntohs(g->port);
+                node->pport = ntohs(g->pport);
```

Comment:
Probably not a breaking change, but if someone did have a work around for handling the plaintext port, when they upgrade we will start gossiping the pport of 0 and serving that to TCP clients. We can only trust that this value is set when we know we're talking with a 6.2 node or something. A very defensive option here would be to add a flag.

Reply:
Actually a node's pport is 0, we don't send it to clients. We send the TLS port to plaintext clients, just like a 6.2 node would do. For example, here is redirects. Cluster slots/nodes is similar.

```C
    } else if (error_code == CLUSTER_REDIR_MOVED ||
               error_code == CLUSTER_REDIR_ASK)
    {
        /* Redirect to IP:port. Include plaintext port if cluster is TLS but
         * client is non-TLS. */
        int use_pport = (server.tls_cluster &&
                         c->conn && connGetType(c->conn) != CONN_TYPE_TLS);
        int port = use_pport && n->pport ? n->pport : n->port;
        addReplyErrorSds(c,sdscatprintf(sdsempty(),
            "-%s %d %s:%d",
            (error_code == CLUSTER_REDIR_ASK) ? "ASK" : "MOVED",
            hashslot, n->ip, port));
```

Reply:
I see, so we only do that pport check, sounds good.